### PR TITLE
ha: fix bug for get subtask from etcd && add integration test 

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 

there is a bug when get subtask

### What is changed and how it works?

fix it, and add a simple integration test for it

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@lichunzhu @csuzhangxc PTAL

File: dm/worker/server.go [hunk 1]
```diff
@@ -640,21 +640,23 @@ func (s *Server) startWorker(cfg *config.MysqlConfig) error {
...
+	resp, err := s.etcdClient.KV.Get(ectx, key, clientv3.WithPrefix())
+	if err != nil {
+		return err
+	}
+	for _, kv := range resp.Kvs {
+		infos := common.UpstreamSubTaskKeyAdapter.Decode(string(kv.Key))
+		taskName := infos[1]
+		task := string(kv.Value)
+		cfg := config.NewSubTaskConfig()
+		if err := cfg.Decode(task); err != nil {
```
Comment (csuzhangxc) [inline dm/worker/server.go]:
```suggestion
		if err = cfg.Decode(task); err != nil {
```

the same with L656.

File: dm/worker/server.go [hunk 2]
```diff
@@ -640,21 +640,23 @@ func (s *Server) startWorker(cfg *config.MysqlConfig) error {
...
+			return nil
```
Comment (csuzhangxc) [inline dm/worker/server.go]: Do we need to return `err` here? the same with L657.

File: dm/worker/server.go [hunk 3]
```diff
@@ -640,21 +640,23 @@ func (s *Server) startWorker(cfg *config.MysqlConfig) error {
...
+		}
+		if err := w.StartSubTask(cfg); err != nil {
+			return nil
 		}
+		log.L().Info("load subtask successful", zap.String("sourceID", cfg.SourceID), zap.String("name", taskName))
```
Comment (csuzhangxc) [inline dm/worker/server.go]:
```suggestion
		log.L().Info("load subtask successful", zap.String("sourceID", cfg.SourceID), zap.String("task", taskName))
```

Is `task` more meaningful here? just as we did in _syncer.go_. `tcontext.Background().WithLogger(log.With(zap.String("task", cfg.Name), zap.String("unit", "binlog replication")))`.

File: tests/ha/conf/dm-master.toml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Master Configuration.
+
+[[deploy]]
+source-id = "mysql-replica-01"
+dm-worker = "127.0.0.1:8262"
+
+[[deploy]]
+source-id = "mysql-replica-02"
+dm-worker = "127.0.0.1:8263"
```
Comment (csuzhangxc) [inline tests/ha/conf/dm-master.toml]: Can we remove these now? (and maybe we need to remove them for other cases.)

File: tests/ha/run.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
```
Comment (csuzhangxc) [inline tests/ha/run.sh]: Do you forget to add this case in _tests/others_integration.txt_?

File: dm/worker/server.go (hunk 1 above)
Comment (csuzhangxc) [reply dm/worker/server.go]: the same with L656

Comment (WangXiangUSTC) [reply dm/worker/server.go]: done

File: dm/worker/server.go (hunk 2 above)
Comment (WangXiangUSTC) [reply dm/worker/server.go]: yes, need to return error, fixed

File: tests/ha/conf/dm-master.toml (hunk 4 above)
Comment (WangXiangUSTC) [reply tests/ha/conf/dm-master.toml]: all removed

File: tests/ha/run.sh (hunk 5 above)
Comment (WangXiangUSTC) [reply tests/ha/run.sh]: yes, I add it in 34cd486

File: tests/ha/run.sh [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    echo "start DM task"
+    dmctl_start_task
+
+    echo "use sync_diff_inspector to check full dump loader"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+
+    echo "start dm-worker3 and kill dm-worker2"
+    run_dm_worker $WORK_DIR/worker3 $WORKER3_PORT $cur/conf/dm-worker3.toml
+    check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER3_PORT
+    ps aux | grep dm-worker2 |awk '{print $2}'|xargs kill || true
```
Comment (csuzhangxc) [inline tests/ha/run.sh]: Do we need to wait for dm-worker2 to become offline?
I find this in the log of CI: `kill: sending signal to 466 failed: No such process`.

Comment (WangXiangUSTC) [reply tests/ha/run.sh]: 1. I add it in 3721168
2.it is the process of grep

**Comment by lichunzhu** [comment]
Can this commit pass HA integration_test?
You can add
```go
	common.PrettyPrintResponse(resp)
```
in [dm/ctl/master/operate_mysql_worker.go](https://github.com/pingcap/dm/blob/ha-dev/dm/ctl/master/operate_mysql_worker.go#L73) to try it

**Pull request merged**